src/aiokatcp/sensor.py

```diff
@@ -225,16 +257,12 @@ def set_value(
             If the incoming `value` type is not compatible with the sensor's
             core type.
         """
-        if not issubclass(type(value), self._core_type):
-            raise TypeError(
-                f"Value type {type(value)} is not compatible with Sensor type "
-                f"{self.stype} with core type {self._core_type}"
-            )
+        checked_value = self._check_value_type(value)
```

Comment:
I'd just do
```suggestion
        value = self._check_value_type(value)
```
That avoids any risk that you accidentally use `value` when you meant to use `checked_value`.

Reply:
Although with an updated signature for `_check_value_type` I'm not so sure this is a good idea any more. `check_value` will have type `_T` while `value` will have type Any.